crates/miden-objects/src/address/mod.rs

```diff
@@ -0,0 +1,91 @@
+use crate::AddressError;
+use crate::account::{AccountId, AccountStorageMode};
+use crate::note::NoteTag;
+
+/// A user-facing address in Miden.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum Address {
```

Comment:
We should add `#[non_exhaustive]` here so that adding more variants is not a breaking change later on.